File: src/test/java/com/tambapps/math/fourier/filtering/FiltersTest.java
package com.tambapps.math.fourier.filtering;

import static org.junit.Assert.assertEquals;

import com.tambapps.math.carray2d.CArray2D;
import com.tambapps.math.complex.Complex;
import org.junit.Test;

public class FiltersTest {

  @Test
  public void rectangleFilterTest() {
    int N = 10;
    CArray2D array = new CArray2D(N, N);
    for (int i = 0; i < array.getN() * array.getM(); i++) {
      array.set(i, Complex.of(i));
    }

    Filters.rectangle(1, 1, false).apply(array);

    CArray2D expected = array.copy();
    for (int i = 0; i < N; i++) {
      expected.set(i, 0, Complex.ZERO);
      expected.set(0, i, Complex.ZERO);
      expected.set(i, N - 1, Complex.ZERO);
      expected.set(N - 1, i, Complex.ZERO);
    }

    assertEquals("Should be equal", expected, array);
  }

}

File: src/test/kotlin/com/tambapps/fft4j/fourier/filtering/FiltersTest.kt
package com.tambapps.fft4j.fourier.filtering

import com.tambapps.fft4j.complex.Complex
import com.tambapps.fft4j.complex.array2d.CArray2D
import junit.framework.Assert.assertEquals
import org.junit.Test

class FiltersTest {
    @Test
    fun rectangleFilterTest() {
        val N = 10
        val array = CArray2D(N, N)
        for (i in 0 until array.n * array.m) {
            array[i] = Complex.of(i.toDouble())
        }
        Filters.rectangle(1, 1, false).apply(array)
        val expected: CArray2D = array.copy()
        for (i in 0 until N) {
            expected[i, 0] = Complex.ZERO
            expected[0, i] = Complex.ZERO
            expected[i, N - 1] = Complex.ZERO
            expected[N - 1, i] = Complex.ZERO
        }
        assertEquals("Should be equal", expected, array)
    }
}
